Fix caret jumping to end of line on truncation.

MGerrior opened the pull request:
When I was using this, I noticed that as you are typing in the middle of the line, when truncation occurs, your caret is sent to the end of the line, like so:

http://g.recordit.co/QeHxfboMiX.gif

Therefore, this PR does two things. Remove the debounce, because I like instant results, but that's more subjective. Secondly, it stores your caret position before truncating, then truncates, then restores your caret position afterwards, like so:

http://g.recordit.co/N8qVgbrliA.gif

It seems like it would make it easier to use, but I'll let you be the judge of that. I was mostly just kind of bored/curious as to how to fix the caret issue.

Commit:
Fix caret jumping to end of line on truncation.

nealrs commented:
Very cool @MGerrior, appreciate the PR!  Can you please also remove the underscore import & then I'll merge? thx

Commit:
Remove unused Underscore, replace jQuery which is 93.5KB with Zepto,
which is only 9.1KB!

MGerrior commented:
No problem @nealrs, removed the unused underscore. I also noticed the jQuery include right next to it. I figured you really only care about modern browsers. I switched that out with Zepto which is about 1/3 of the size of jQuery (24.4KB vs 93.5KB) and is even smaller when served through the CDN since it is gzipped and only 9.1KB. We're not doing any fancy animations or anything, so there should be no noticeable difference for users.
